src/Http/Client.php

```diff
@@ -44,15 +44,14 @@ public function __construct(
         StreamFactoryInterface $streamFactory = null
     ) {
         $this->baseUrl = $url;
-        $this->apiKey = $apiKey;
         $this->http = $httpClient ?? Psr18ClientDiscovery::find();
         $this->requestFactory = $reqFactory ?? Psr17FactoryDiscovery::findRequestFactory();
         $this->streamFactory = $streamFactory ?? Psr17FactoryDiscovery::findStreamFactory();
         $this->headers = array_filter([
             'User-Agent' => implode(';', array_merge($clientAgents, [Meilisearch::qualifiedVersion()])),
         ]);
-        if (null != $this->apiKey) {
-            $this->headers['Authorization'] = sprintf('Bearer %s', $this->apiKey);
+        if (null !== $apiKey) {
```

Comment:
@brunoocasali unfortunately this made some troubles for environment keys that are passed in as empty string as mentioned here: 

https://github.com/laravel/scout/issues/769

Reply:
Imho we should add phpdoc `@param non-empty-string|null $apiKey`

Reply:
@norkunas I'm posting the issue description now for tracking